From f43b8dc72418a4bf57df56b0d61ee423b5cb206c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:28:13 -0400
Subject: [tools] perf/traceevent: Fix conversion of pointer to integer of different size

Message-id: <1408462094-14194-325-git-send-email-jolsa@redhat.com>
Patchwork-id: 88306
O-Subject: [PATCH RHEL7.1 BZ1131394 324/325] tools lib traceevent: Fix conversion of pointer to integer of different size
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 6b5fa0ba4f85a8499287aefaf3f1375450c40c6d
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Nov 19 16:14:51 2013 -0300

description
===========
gcc complaint on 32-bit system:

  /home/acme/git/linux/tools/lib/traceevent/event-parse.c: In function 'eval_num_arg':
  /home/acme/git/linux/tools/lib/traceevent/event-parse.c:3468:9: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]

This is because the eval_num_arg returns everything as an 'unsigned long long',
so it converts a void pointer to a wider integer, fix it by converting the void
pointer to an integer of the same size, 'unsigned long', before casting it to
'unsigned long long'.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.c b/tools/lib/traceevent/event-parse.c
index 8a5b65d..217c82e 100644
--- a/tools/lib/traceevent/event-parse.c
+++ b/tools/lib/traceevent/event-parse.c
@@ -3465,7 +3465,7 @@ eval_num_arg(void *data, int size, struct event_format *event, struct print_arg
    * is in the bottom half of the 32 bit field.
    */
   offset &= 0xffff;
-  val = (unsigned long long)(data + offset);
+  val = (unsigned long long)((unsigned long)data + offset);
   break;
  default: /* not sure what to do there */
   return 0;
-- 
1.7.1